reference/zrc6.scilla

```diff
@@ -0,0 +1,984 @@
+  send msgs
+end
+
+(* Pauses the contract. Use this when things are going wrong ('circuit breaker'). *)
+(* @Requirements: *)
+(* - The contract should not be paused. Otherwise, it should throw `PausedError` *)
+(* - `_sender` should be the contract owner. Otherwise, it should throw `NotContractOwnerError` *)
+transition Pause()
+  RequireNotPaused;
+  RequireContractOwner;
+
+  is_paused := true;
+  e = {
+    _eventname: "Pause";
+    initiator: _sender;
+    is_paused: true
+  };
+  event e;
+  msg_to_sender = {
+    _tag: "ZRC6_PauseCallback";
+    _recipient: _sender;
+    _amount: Uint128 0;
+    is_paused: true
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Unpauses the contract. *)
+(* @Requirements: *)
+(* - The contract should be paused. Otherwise, it should throw `NotPausedError` *)
+(* - `_sender` should be the contract owner. Otherwise, it should throw `NotContractOwnerError` *)
+transition Unpause()
+  paused <- is_paused;
+  match paused with
+  | True =>
+  | False =>
+    error = NotPausedError;
+    Throw error
+  end;
+  RequireContractOwner;
+
+  is_paused := false;
+  e = {
+    _eventname: "Unpause";
+    initiator: _sender;
+    is_paused: false
+  };
+  event e;
+  msg_to_sender = {
+    _tag: "ZRC6_UnpauseCallback";
+    _recipient: _sender;
+    _amount: Uint128 0;
+    is_paused: false
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Sets `to` as the contract owner candidate. *)
+(* To reset `contract_owner_candidate`, use `zero_address`. *)
+(* i.e., `0x0000000000000000000000000000000000000000` *)
+(* @param: to - contract owner candidate address  *)
+(* @Requirements: *)
+(* - `_sender` should be the contract owner. Otherwise, it should throw `NotContractOwnerError` *)
+(* - `_sender` should not be `to`. Otherwise, it should throw `SelfError` *)
+transition SetContractOwnerCandidate(to: ByStr20)
+  RequireContractOwner;
+  RequireNotSelf to _sender;
+  contract_owner_candidate := to;
+  e = {
+    _eventname: "SetContractOwnerCandidate";
+    initiator: _sender;
+    to: to
+  };
+  event e;
+  msg_to_sender = {
+    _tag: "ZRC6_SetContractOwnerCandidateCallback";
+    _recipient: _sender;
+    _amount: Uint128 0;
+    to: to
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Sets `contract_owner_candidate` as the contract owner. *)
+(* @Requirements: *)
+(* - `_sender` should be the contract owner candidate. Otherwise, it should throw `NotContractOwnerCandidateError` *)
+transition AcceptContractOwnership()
+  candidate <- contract_owner_candidate;
+
+  is_candidate = builtin eq _sender candidate;
+  match is_candidate with
+  | False =>
+    error = NotContractOwnerCandidateError;
+    Throw error
+  | True =>
+    contract_owner := _sender;
+    contract_owner_candidate := zero_address;
+
+    e = {
+      _eventname: "AcceptContractOwnership";
+      initiator: _sender;
+      contract_owner: _sender;
+      contract_owner_candidate: zero_address
```

Comment:
This may not be necessary

Reply:
I agree since it is always zero address.
I should remove it from the callback too.